src/exe/Main.hs

```diff
@@ -257,24 +304,15 @@ initPMState opts = PMState { nextId     = 0
 --   module name. Otherwise, the base name of the input file is used.
 makeOutputFileName
   :: FilePath     -- ^ The name of the input file.
-  -> HSE.Module l -- ^ The module to make the output file name of.
+  -> Maybe String -- ^ The name of the module to make the output file name of.
   -> FilePath     -- ^ The name of the output file.
-makeOutputFileName inputFile inputModule = outputFileName <.> "hs"
+makeOutputFileName inputFile modName = outputFileName <.> "hs"
  where
   -- | The output file name without file extension.
   outputFileName :: FilePath
-  outputFileName = maybe (takeBaseName inputFile)
-                         (joinPath . splitOn ".")
-                         (moduleName inputModule)
+  outputFileName =
+    maybe (takeBaseName inputFile) (joinPath . splitOn ".") modName
 
-  -- | Gets the module name of the given module.
-  --
-  --   Returns @Nothing@ if there is no module header or the module type is not
-  --   supported.
-  moduleName :: HSE.Module l -> Maybe String
-  moduleName (HSE.Module _ (Just moduleHead) _ _ _) = case moduleHead of
-    (HSE.ModuleHead _ (HSE.ModuleName _ modName) _ _) -> Just modName
-  moduleName _ = Nothing
 
 -- | Pretty prints the given Haskell module.
 prettyPrintModule :: HSE.Module HSE.SrcSpanInfo -> String
```

Comment:
This function should be named after or parameterized over the front ends as well.